FEXCore/Source/Interface/Core/ArchHelpers/Arm64Emitter.cpp

```diff
@@ -822,6 +824,10 @@ void Arm64Emitter::FillStaticRegs(bool FPRs, uint32_t GPRFillMask, uint32_t FPRF
     }
   }
 
+  if (FillP2) {
```

Comment:
Perhaps you can drop this and change to a simple sve supported check, as is it'll not work for the FillSRA in dispatcher and a single ptrue is cheap enough not to matter

Reply:
But if we always fill `p2` when SVE is supported this whole optimization makes no longer sense from a size pov. If we do it without actually needing to use `p2` for the SVE store optimization then we are definitely pessimizing things.

Reply:
Spills are rare enough that a ptrue certainly isn't going to change anything, the opt still works within code blocks which is where it matters as they can be much botter

Reply:
Humm, I still didn't manage to get SRA to work. The RA algo is still allocating the wrong register to the predicate register, i.e. p5. It should be p2, which is the SRA register I selected to use in DecodeSRAReg but it's not working. First time I am looking at this code so slightly confused about this.

At the same time, i am wondering if this is necessary. What's the advantage of using SRA over just hardcoding p2 usage in the instructions, like done with `PRED_TMP_16B` for example?

Reply:
> Humm, I still didn't manage to get SRA to work. The RA algo is still allocating the wrong register to the predicate register, i.e. p5. It should be p2, which is the SRA register I selected to use in DecodeSRAReg but it's not working. First time I am looking at this code so slightly confused about this.
> 
> At the same time, i am wondering if this is necessary. What's the advantage of using SRA over just hardcoding p2 usage in the instructions, like done with `PRED_TMP_16B` for example?

I don't think there are any, that would work fine too you'd just need to modify the store op to have some way to indicate that and you can drop LoadSVEOptPredicate and rely on it being set in FillSpecialRegs as PRED_TMP_16B is (you probably want to keep InitPredicate for other future users)

Reply:
(can drop this as it's called in FillSpecialRegs)

Reply:
Right - thanks.